### PR TITLE
Create T012_M3_batch_ui.py

### PR DESCRIPTION
**Pull request by Hey1121**
#10

### CODE REVIEW

**Review by InochiFumetsu — CHANGES_REQUESTED** [review]
You need to upload the batch file too!

**Review by Jackinator27 — APPROVED** [review]

File: T012_M3_batch_ui.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#by Esteban Heidrich SID: 101267959
+
+import T012_M1_load_data
+import T012_M2_sort_plot
+import T012_M3_optimization
+def batch_ui():
+    command_list = []
+    print("Please input the name of file where the commands are sotred")
```
Comment (InochiFumetsu) [inline T012_M3_batch_ui.py:8]:
```suggestion
    print("Please input the name of file where the commands are stored")
```

File: T012_M3_batch_ui.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                print("There is a command error in the file")
+                break
+        elif command_list[0].lower() == "w":
+            min_result = T012_M3_optimization.minimum(
+                loaded_data, command_list[1])
+            print("Lowest grade average for",
+                  command_list[1], "is", min_result)
+        elif command_list[0].lower() == "b":
+            max_result = T012_M3_optimization.maximum(
+                command_list[1], loaded_data)
```
Comment (InochiFumetsu) [inline T012_M3_batch_ui.py:54]:
```suggestion
                loaded_data, command_list[1])
```

**Review by InochiFumetsu — CHANGES_REQUESTED** [review]
The second code suggestion needs to go through (the profs said the dict/att arguments for the optimization functions could go either way but was supposed to be <dict>, <attr> (so I swapped them around in my code))

File: T012_M3_batch_ui.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            except:
+                print("There is a command error in the file")
+                break
+        elif command_list[0].lower() == "s":
+            sorted_data = T012_M2_sort_plot.sort_students_bubble(
+                loaded_data, command_list[1])
+            if command_list[2] == 'Y':
+                print("Here is your sorted data")
+                print(sorted_data)
+            elif command_list[2] == 'N':
```
Comment (InochiFumetsu) [inline T012_M3_batch_ui.py:33]: I didn't read the function / module requirements for your guys' code -- but I notice you don't accept lower-case 'y' or 'n' (or 'yes' or 'no' or 'yEs' etc.) -- does that matter or?

**Review by InochiFumetsu — APPROVED** [review]

File: T012_M3_batch_ui.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                    print("Data Loaded")
+            except:
+                print("There is a command error in the file")
+                break
+        elif command_list[0].lower() == "s":
+            sorted_data = T012_M2_sort_plot.sort_students_bubble(
+                loaded_data, command_list[1])
+            if command_list[2] == 'Y':
+                print("Here is your sorted data")
+                print(sorted_data)
```
Comment (InochiFumetsu) [inline T012_M3_batch_ui.py:32]:
```suggestion
                for i in iter(sorted_data):
                    print(i, ":", sep="")
                    for j in range(len(sorted_data[i])):
                        print("\t", sorted_data[i][j], sep="")
```

Comment (InochiFumetsu) [reply T012_M3_batch_ui.py:32]: not important -- same suggestion I made to Jack -- just displays nicer. Try it in your local version, see if you like.

File: T012_M3_batch_ui.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        elif command_list[0].lower() == "w":
+            min_result = T012_M3_optimization.minimum(
+                loaded_data, command_list[1])
+            print("Lowest grade average for",
+                  command_list[1], "is", min_result)
+        elif command_list[0].lower() == "b":
+            max_result = T012_M3_optimization.maximum(
+                loaded_data, command_list[1])
+            print("Highest grade average for",
+                  command_list[1], "is", max_result)
```
Comment (InochiFumetsu) [inline T012_M3_batch_ui.py:56]:
```suggestion
                  command_list[1], "is", max_result)
          file_contents.close()
```

**Review by Jackinator27 — APPROVED** [review]

**Review by InochiFumetsu — APPROVED** [review]

**Review by InochiFumetsu — APPROVED** [review]

**Review by Jackinator27 — APPROVED** [review]

**Review by MilanxDjordjevic — APPROVED** [review]

**Pull request merged**